numba/targets/base.py

```diff
@@ -487,22 +487,18 @@ def imp(context, builder, typ, val):
         if isinstance(typ, types.Module):
             # Implement getattr for module-level globals.
             # We are treating them as constants.
-            # XXX We shouldn't have to retype this
```

Comment:
Normally, the attribute type has already been deduced during the type inference phase. This method is called during lowering, so it should already have the information at hand (but due to the way it is called, it doesn't, AFAIK).